src/js/binaryen.js-post.js

```diff
@@ -541,8 +541,8 @@ function wrapModule(module, self) {
     'set': function(index, value) {
       return Module['_BinaryenLocalSet'](module, index, value);
     },
-    'tee': function(index, value) {
-      return Module['_BinaryenLocalTee'](module, index, value);
+    'tee': function(index, value, type) {
+      return Module['_BinaryenLocalTee'](module, index, value, type);
```

Comment:
Can we make it so that if `type` is not specified, the type of `value` is used - to make this backwards compatible?

Reply:
That would only work in non-reference types. The purpose of this change is to prevent to use the value type for `local.tee`'s type in the first place because of subtypes... Wouldn't that be easy to be used incorrectly?

Reply:
Mostly thinking in terms of existing projects that don't use reference types yet, where having to change every single `local.tee` might be quite some work, even though these don't use reference types (yet). Perhaps only make if backwards compatible for i32, i64, f32, f64?

Reply:
I am still inclined to change the signature to be consistent; making the old signature only work for old number types seem a little unnatural and also adds checking overhead to every `local.tee` call. I don't know much about the cost of changing existing code, but doesn't all existing code have to undergo significant changes after #2510 lands anyway..? WDYT? cc @kripken

Reply:
I would like to make the change as less confusing as possible for people with existing code. It seems like they would not provide the parameter, so it would be coerced from undefined to 0, which is `none`. Then they'd get a validation error later, which might not be that obvious?

Automatically converting the 0 to the proper type is one option, silently - IIUC @dcodeIO that's what you suggest?

Another option is to give an immediate error, `if (type === undefined) throw Error('tee() must receive the type as a fourth parameter (this was an API change)`);` perhaps? A benefit to doing it that way is people will know to update their code, so it won't keep using the old API forever.

Reply:
What do you mean by
> Automatically converting the 0 to the proper type is one option
?

Reply:
I meant what @dcodeIO said, at least as I understood it:

> Can we make it so that if type is not specified, the type of value is used - to make this backwards compatible?

If `type` is not provided, use the type of the `value` instead.

Reply:
What do you think @decodeIO? I like the idea of requiring `type` parameter in all `local.tee`, and explicitly error out if `type` is undefined. Allowing untyped API only for the number types doesn't seem to be very consistent. Can we land this?

Reply:
What do you think @dcodeIO? I like the idea of requiring `type` parameter in all `local.tee`, and explicitly error out if `type` is undefined. Allowing untyped API only for the number types doesn't seem to be very consistent. Can we land this?

Reply:
Sure, go ahead :). Should be fine with other breaking changes already landed.